### ISSUE TITLE
Promxy returning empty results intermittently

### ISSUE THREAD

**Issue by ottoyiu** [jacksontj/promxy#362]
Perhaps related to #355.

We are using Promxy as a kubernetes cluster-wide aggregator of multiple discrete Prometheus deployments within the cluster. 

Promxy is currently configured to retrieve metrics from two separate server groups via promql using `/api/v1/query_range` rather than remote reads (ie. `remote_read: False`).

Each server group is discovering a service that points to Thanos Querier Frontends (a caching proxy infront of Thanos Querier).

In this example, I have two prometheus deployments that I want to aggregate using Promxy. One deployment is `example` in the `dev-jsiaw` namespace, and the other is `oyiu-pipe` in the `oyiu-pipe` namespace.

Config:
```
  config.yaml: |
    global:
      evaluation_interval: 5s
      external_labels:
        source: promxy
    promxy:
      server_groups:
      - kubernetes_sd_configs:
        - namespaces:
            names:
            - dev-jsiaw
          role: endpoints
        labels:
          pipeline_name: example
          pipeline_namespace: dev-jsiaw
        relabel_configs:
        - action: keep
          regex: ^example-thanos-query-frontend$
          separator: ;
          source_labels:
          - __meta_kubernetes_endpoints_name
        - action: keep
          regex: ^http$
          separator: ;
          source_labels:
          - __meta_kubernetes_endpoint_port_name
        remote_read: false
        ignore_error: true
      - kubernetes_sd_configs:
        - namespaces:
            names:
            - oyiu-pipe
          role: endpoints
        labels:
          pipeline_name: oyiu-pipe
          pipeline_namespace: oyiu-pipe
        relabel_configs:
        - action: keep
          regex: ^oyiu-pipe-thanos-query-frontend$
          separator: ;
          source_labels:
          - __meta_kubernetes_endpoints_name
        - action: keep
          regex: ^http$
          separator: ;
          source_labels:
          - __meta_kubernetes_endpoint_port_name
        ignore_error: true
        remote_read: false
    remote_write: []
    rule_files:
    - '*rule'
```

When requesting range queries, Promxy returns an empty result even though the Thanos Query Frontend returns a proper result. This occurs 6/10 times. It happens even more frequent on larger range queries.

Example query: go_memstats_gc_cpu_fraction{pipeline_name="example"}

Promxy:
```
{

    "status": "success",
    "data": {
        "resultType": "matrix",
        "result": [ ]
    }

}
```

vs Thanos Query Frontend:
```

{

    "status": "success",
    "data": {
        "resultType": "matrix",
        "result": [
            {
                "metric": {
                    "__name__": "go_memstats_gc_cpu_fraction",
                    "instance": "10.79.1.124:10902",
                    "job": "pods-nginx",
                    "kubernetes_namespace": "dev-jsiaw",
                    "kubernetes_pod_name": "example-thanos-query-frontend-86dcdf565d-lp9ms",
                    "pipeline_name": "example",
                    "pipeline_namespace": "dev-jsiaw",
                    "receive_replica": "example-thanos-receiver-0",
                    "tenant_id": "dev-jsiaw-example"
                },
                "values": [
                    [
                        1604612203,
                        "0.000053259467817038186"
                    ],
                  ...
}
```

Diving into the code, I've added debugging messages and see that the PromAPIV1 and MultiAPI `QueryRange` calls do succeed , and contain the proper result that is returned by the Thanos Query Frontend. However, the returned results is empty from Promxy's api/v1/query_range call.

 ProxyStorage.NodeReplacer also is seen to properly getting the results back from `state.client.QueryRange` and set the Series properly:
https://github.com/jacksontj/promxy/blob/9012d2833f6d1960a1406bd531f716c498036b3f/pkg/proxystorage/proxy.go#L515-L536

At this point, I'm lost in the code to dive further. Any help would be appreciated it!

**Comment by dgrizzanti** [comment]
@ottoyiu just stumbled on your issue as I've observing the same behavior (in my case I'm pointing to 2 Prometheus instances in 1 group and 2 Thanos query instances in the other group). See about the same ratio of missing data for the Thanos group as you.

Out of curiosity, what version are you running?

**Comment by ottoyiu** [comment]
Glad to here that we're not the only ones. Running Thanos Query 0.16.0; older versions also exhibit the same issue. For the time being, we've moved away from promxy and are nesting the thanos queriers to achieve similar results.

**Comment by dgrizzanti** [comment]
We're running Thanos Query 0.14.0 but I tested today with the latest version of promxy (`v0.0.62`) and I don't see the issues any longer. Previously we were running `v0.0.57`, so maybe check that?

**Comment by jacksontj** [comment]
Given that this is fixed in later versions of promxy this _seems_ to be another report of https://github.com/jacksontj/promxy/pull/304 -- fixed in https://github.com/jacksontj/promxy/pull/304 -- first landed in v0.0.59).

**Comment by jacksontj** [comment]
There hasn't been any update on this issue in a while (and we believe we have this resolved already in newer versions), so I'm going to close this out. If this is still an issue please feel free to re-open!
